src/Groups/GAPGroups.jl

```diff
@@ -2107,18 +2107,8 @@ x*y^4
 ```
 """
 function map_word(g::Union{PcGroupElem, SubPcGroupElem}, genimgs::Vector; genimgs_inv::Vector = Vector(undef, length(genimgs)), init = nothing)
-  G = parent(g)
-  Ggens = gens(G)
-  if length(Ggens) == 0
-    return init
-  end
-  gX = GapObj(g)
-
-  if GAPWrap.IsPcGroup(GapObj(G))
-    l = GAP.Globals.ExponentsOfPcElement(GAP.Globals.FamilyPcgs(GapObj(G)), gX)
-  else  # GAP.Globals.IsPcpGroup(GapObj(G))
-    l = GAP.Globals.Exponents(gX)
-  end
+  length(gens(parent(g))) == 0 && return init
```

Comment:
```suggestion
  ngens(parent(g)) == 0 && return init
```
If `init` is not provided this returns `nothing` -- is that what we want? I'd expect an error based on how other similar `reduce` variants work.

Reply:
Yes, the other `map_word` methods throw an exception if there are no generators and if `init` is not given.